server/player/player.go

```diff
@@ -667,8 +678,9 @@ func (d *Data) TakeItem(id int32, slot int16, amount int16, invID byte, db *sql.
 
 }
 
-func (d *Data) updateItemStack(item item.Data, db *sql.DB) {
-	item.UpdateStack(db)
+func (d Data) updateItemStack(item item.Data, db *sql.DB) {
+	item.Save(db, d.id)
+	d.updateItem(item)
```

Comment:
Passing the database ptr around everywhere is getting kind of messy, I think it would suit being a global of it's own package (done as another PR)?

Reply:
I agree completely. It would be best to have the database as its own package. I think that would also help with some syncing issues as well. I also agree that it is something that would need to be a separate PR also it can be tracked easier that way.